clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -87,8 +88,11 @@ public boolean hasInitialResponse() {
                     if (challenge != null && challenge.length != 0)
                         throw new SaslException("Expected empty challenge");
                     callbackHandler().handle(new Callback[] {callback});
+                    SaslExtensionsCallback extensionsCallback = parseCustomExtensions();
```

Comment:
Please update Javadoc for the class to also state that the instance of {@code AuthenticateCallbackHandler} can optionally handle an instance of {@link SaslExtensionsTokenCallback} to return any extensions generated by the {@code login()} event on the {@code LoginContext}.